### PR TITLE
241: Close gaps between paragraphs in preview

### PR DESCRIPTION
**Pull request by tamsingreen**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
#241 

**Describe what the PR does**    
- [x] add data attribute to words in transcript with sourceIndex (in Word.js)
- [x]  add sourceIndex to element object
- [x]  can delete and move clips around as before
- [x]  on playback, check if clip source and index are continuous and treat as one clip if so (export out of scope)
- [x] adds test for compilePlaylist function (this PR also fixes test config in #215 but hasn't touched the Husky config)
Additionally refactors some of the functions in programmeScriptContainer.

**State whether the PR is ready for review or whether it needs extra work**    
RFR
## User Story / Context
|As a ...|I want ...|So that ...|
|-|-|-|
|Producer/editor|Sequential paragraphs to playback as one continuous clip|So there are no unnecessary edits|

## Acceptance Criteria
- Test passes (compilePlaylist.test.js)
- Create a paper edit (easier to see edits with a video than audio). Add a few continuous and non-continuous paragraphs. Check playback that there are no edits between the continuous clips. Move or delete some clips, check still handled correctly.

### CODE REVIEW

File: src/Components/PaperEditor/ProgrammeScriptContainer/index.js [hunk 1]
```diff
@@ -148,6 +151,7 @@ const ProgrammeScriptContainer = (props) => {
     }
   }, [ elements, paperEdits ]);
 
+  //when paperEdits are updated, update transcripts
```
Comment (allishultes) [inline src/Components/PaperEditor/ProgrammeScriptContainer/index.js:154]: Thank you for adding these comments, I think they make it much quicker to understand what's happening with the code 💯

File: src/Components/PaperEditor/ProgrammeScriptContainer/utils/compilePlaylist.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      let startTime = 0;
+      const playlistLength = prevResult.length;
+      if (playlistLength > 0) {
+        const lastPlaylistItem = prevResult[playlistLength - 1];
+        if (paperEditsAreContinuous(lastPlaylistItem, paperEdit)) {
+          const updatedPlaylistItem = {
+            ...lastPlaylistItem,
+            duration: paperEdit.end - lastPlaylistItem.sourceStart,
+            lastSourceParagraphIndex: paperEdit.sourceParagraphIndex
+          };
```
Comment (allishultes) [inline src/Components/PaperEditor/ProgrammeScriptContainer/utils/compilePlaylist.js:33]: This is so nice and readable, I love it

**Review by allishultes — APPROVED** [review]
Had a play with this and it looks really good! Plus soo exciting we have tests working now :)

Should we open a ticket for adding this feature to the export functionality?

**Review by sarahrainbow — APPROVED** [review]
wow this is a lot of work – looks amazing and seems to work great!

**Pull request merged**